openwisp_controller/config/static/config/js/widget.js

```diff
@@ -1,6 +1,51 @@
 'use strict';
 (function ($) {
     var inFullScreenMode = false,
+        prevDefaultValues = {},
+        defaultValuesUrl = window.location.origin + '/admin/config/device/get-template-default-values/',
+    removeDefaultValues = function(contextValue, defaultValues={}) {
+        // remove obsulate default values from contextValue
+        Object.keys(prevDefaultValues).forEach(function (key) {
+            if (!defaultValues.hasOwnProperty(key) &&
+                prevDefaultValues[key] === contextValue[key]) {
+                delete contextValue[key];
+            }
+        });
+        return contextValue;
+    },
+    updateContext = function (defaultValues={}) {
+        var contextField = $('#id_config-0-context');
+        if (contextField.length) {
+            var contextValue = JSON.parse(contextField.val());
+            // add default values to contextValue
+            Object.keys(defaultValues).forEach(function (key) {
+                if (!contextValue.hasOwnProperty(key)) {
+                    contextValue[key] = defaultValues[key];
+                }
+            });
+
+            contextField.val(JSON.stringify(
+                removeDefaultValues(contextValue, defaultValues),
+                null,
+                4
+            ));
+            prevDefaultValues = JSON.parse(JSON.stringify(defaultValues));
+            $('.flat-json-toggle-textarea').trigger('click');
```

Comment:
This is where i got stuck for hours :smile: . The objective of this two lines is to toggle the `configuration variable` widget so that the value that was filled up at the `raw json editing` can appear at the UI too. It works well but behaves funny on the `device change page`. The flip is done half way.